---
title: "Coral holobiont physiology under ocean acidification and warming"
date: "*Last run on `r format(Sys.time(), '%d %B %Y')`*"
output: 
  html_document:
    theme: simplex
    toc: yes
    toc_float: yes
  pdf_document:
    toc: yes
---

<style>
  h2{color: teal !important}
  h1{color: navy !important}
  body{background-color: gray95 !important}
</style>

<br/>

<center>
**This is a working document for all the coral physiology data manipulation, analysis, and visualization portion of the physiology/gene expression manuscript. The document includes all figures, tables, supplemental materials, methods, and results for the physiology portion of the manuscript. Each major analysis is separated by the tabs on the left.** 
</center>

<br/>

---

<span style="color: red;">**Current plan of attack:**</span>

1. ~~Compile figures/tables, results, methods~~ -- *done!*
2. Approval of these materials (Sarah) -- *now*
    + Approval from Karl/Justin? -- *do we do this before drafting?*
3. Update introduction -- *goal of June 1*
4. Update discussion -- *goal of July 1*
5. Send to coauthors -- *goal of August 1*

---

<br/>

```{r setup, include=FALSE}

knitr::opts_chunk$set(echo = FALSE, warning = FALSE, message = FALSE, options(knitr.kable.NA = ''))
options(warn = -1)

```

```{r packages, include=FALSE}

#library("devtools")
#install_github("vqv/ggbiplot") # need this to download ggbiplot if not already installed

## Packages to load
library(knitr)
library(readr)
library(broom)
library(ggplot2)
library(dplyr)
library(ggbiplot)
library(tidyr)
library(corrgram)
library(openxlsx)
library(plotly)
library(tidyverse)
library(vegan)
library(shiny)
library(Rmisc)
library(cowplot)
library(ggfortify)
library(finalfit)
library(kableExtra)
library(readr)
library(lmerTest)
library(vroom)
library(ggpubr)
library(magick)
library(Hmisc)
library(corrplot)
library(gridGraphics)
library(grid)
library(RColorBrewer)
library(wesanderson)
library(performance)
library(MASS)
library(png)
library(car)

```

```{r set formatting and labels, include=FALSE}

## Set some standards and units

# dodge width
dodge=position_dodge(width=0.3)
dodge2=position_dodge(width = 0.6)
jitter=position_jitter(width=0.1)

# set figure theme
theme_set(theme_pubr())

# set parameter labels
Tlab<-"Temperature (Â°C)"
alab<-expression(paste(italic("p"),"CO" [2]~ "("*mu,'atm)'))
dlab<-expression(paste("Cell density (10"^6,"cells cm"^-2,")"))
plab<-expression(paste("Total protein (mg cm"^-2,")"))
calab<-expression(paste("Carbohydrate (mg cm"^-2,")"))
clab<-expression(paste("Chlorophyll a ("*mu,"g cm"^-2,")"))
rlab<-expression(paste("Calcification rate (mg cm"^-2~"day"^-1,")"))
llab<-expression(paste("Total Lipid (mg cm"^-2,")"))
hlab<-expression(paste("Total Host (mg cm"^-2,")"))

# set the current date
date <- Sys.Date() 

```

```{r original dataframe adjustment, include=FALSE}

#df <- read_csv("Data/Raw_data/phys_all_21Oct19.csv") # read in full dataframe
#den_count <- read_csv("/Users/colleen/Dropbox/Grad school/UNC_PhD_Folders/Boston_Experiment/Physiology/Symbiont #Density/density_ForAddition_23Mar21.csv")
#
#test <- merge(df, den_count, by = "coral")
#test$X1 <- NULL
#test$X1_1 <- NULL 
#
#write.csv(test, "Data/Raw_data/phys_all_23March2021.csv")

df <- read_csv("Data/Raw_data/phys_all_23March2021.csv") # read in full dataframe
df$X1 <- NULL

# rename a couple columns
names(df)[38]<-"carb"
names(df)[39]<-"lipid"
names(df)[17]<-"sum"
names(df)[34]<-"red" 

# set columns as factors
df$ftemp <- as.factor(df$ftemp)
df$fpco2 <- as.factor(df$fpco2)
df$colony <- as.factor(df$colony)
df$species <- factor(df$species, levels = c("S", "P", "A", "T")) # and reorder these

# replace some pCO2 values with ones we will use moving forward
df$fpco2 <- gsub("2800", "3300", df$fpco2)
df$fpco2 <- gsub("280", "300", df$fpco2)
df$fpco2 <- gsub("400", "420", df$fpco2)
df$fpco2 <- gsub("700", "680", df$fpco2)

# reorder factors for pCO2 and temp
df$fpco2 <- factor(df$fpco2, c("420", "T0", "300", "680", "3300"))
df$ftemp <- factor(df$ftemp, c("28", "T0", "31"))

# calculate phys parameters
df$den <- df$den / 1000000 # adjust symbiont density to display 10^6 cells
df$host <- df$pro + df$carb + df$lipid # calculate total host energy reserves (sum of carb, protein, lipids)
df$treat[df$T0_T90 == "T0"] <- "T0" # replace T0 'treat' with T0 text
df$treat <- factor(df$treat, c("T0", "288_28", "311_31", "447_28", "405_31", "673_28", "701_31", "3285_28", "3309_31")) # adjust treatment levels

# add a new treat column for plotting (replacing actual treatment with number for better plotting - see plot XX)
df$treat2 <- df$treat
df$treat <- gsub("T0", 2, df$treat)
df$treat <- gsub("288_28", 4, df$treat)
df$treat <- gsub("311_31", 4, df$treat)
df$treat <- gsub("447_28", 6, df$treat)
df$treat <- gsub("405_31", 6, df$treat)
df$treat <- gsub("673_28", 8, df$treat)
df$treat <- gsub("701_31", 8, df$treat)
df$treat <- gsub("3285_28", 10, df$treat)
df$treat <- gsub("3309_31", 10, df$treat)
df$treat <- as.numeric(df$treat) # convert 'treat' column to numerics (again, this is for plotting only)

# inverse of colours (so lower colour depicts more bleached coral)
df$sum <- (df$sum * -1)
df$red <- (df$red * -1)
df$blue_bw5 <- (df$blue_bw5 * -1)
df$green_bw5 <- (df$green_bw5 * -1)

## modify dataframe for simplified version
df2 <- df[,c(1:13, 41, 14:16, 37:40, 42:43, 17)] # select columns of interest only
df2 <- gather(df2, param, value, den:sum) # make column of parameter and value
df2$param <- factor(df2$param, levels = c("pro", "carb", "lipid", "den", "chla", "host", "count")) # reorder parameters
df2$species <- revalue(x = df2$species, c("S" = "SSID", "P" = "PSTR", "A" = "PAST", "T" = "UTEN")) # rename the species codes
df2 <- subset(df2, species != "UTEN") # remove UTEN (omitted due to high mortality in some treatments)

## remote T0 samples from dataframe for models
df2_T90 <- subset(df2, T0_T90 == "T90")

## add treatment column
df2_T90$treat2 <- paste0(df2_T90$fpco2, df2_T90$temp)

```

```{r bootstrap model setup}

## Function to remove rows with any missing values
completeFun <- function(data, desiredCols) {
  completeVec <- complete.cases(data[, desiredCols])
  return(data[completeVec, ])
}

## Performing the parametric bootstrapping of the model:
bootnum = 1500 # set number of iterations (we used 2000) between 999 and 9999
seed = 3 #seed to make results replicatable (our seed was 7)
set.seed(3)

## This is a bootstrap function that will sample WITH replacement (explained step by step within the function)
bootFUN <- function(model, newdata) {
  nr <- nrow(model$data) # count number of rows in the model (# of observations)
  data <- model$data # pull data from model
  data <- data[sample(1:nr, size = nr, replace = TRUE), ] # random sample of numbers from 1 - nr (# of rows) to select data from
  update <- update(model, data = data) # rerun the model using the 'new' dataset from the random smapling above
  predict(update, newdata = newdata, type = "response") # predicts response variable with model and updated data 
}

#bootFUN <- function(model, newdata) {
#  nr <- nrow(model$data) # count number of rows in the model (# of observations)
#  data <- model$data # pull data from model
#  
#  colnames(newdata)
#
#  data %>% group_by() %>% sample_frac(.5)
#  
#  #data <- data[sample(1:nr, size = nr, replace = TRUE), ] # random sample of numbers from 1 - nr (# of rows) to select data from
#  #update <- update(model, data = data) # rerun the model using the 'new' dataset from the random smapling above
#  #predict(update, newdata = newdata, type = "response") # predicts response variable with model and updated data 
#}

```

```{r create dataframe per parameter}

## Create a forloop to subset the data by each parameter and save as individual dataframes 

# make a list of parameter names
param_list <- levels(df2_T90$param)

# forloop for dataframes
for (p in 1:length(param_list)) {
  param_select <- param_list[p]
  df_subset <- subset(df2_T90, param == param_select) # subset the dataframe for only one parameter
  df_subset <- completeFun(df_subset, "value") # run function to remove any missing values
  df_subset$value <- as.numeric(df_subset$value)
  df_subset$ftemp <- droplevels(df_subset$ftemp)
  df_subset$fpco2 <- droplevels(df_subset$fpco2)
  df_subset$colony <- droplevels(df_subset$colony)
  df_subset$species <- droplevels(df_subset$species)
  assign(paste(param_select, "mod_df", sep = "_"), df_subset) # assign the data to dataframe named for each parameter 
}

```

```{r T90 df setup}

df_90 <- subset(df, T0_T90 == "T90") 
df_90 <- subset(df_90, species != "T") 
df_90 <- df_90[,-c(18:33)]
df_90 <- completeFun(df_90, "den")
df_90 <- completeFun(df_90, "host")
df_90 <- completeFun(df_90, "lipid")
df_90 <- completeFun(df_90, "carb")
df_90_l <- gather(df_90, param, value, 14:23)

## specific species dataframes
s_df <- subset(df_90, species == "S") 
p_df <- subset(df_90, species == "P") 
a_df <- subset(df_90, species == "A") 

```

```{r add in the SSID dominant symbionts}

symbs <- read.csv("Data/Raw_data/domSSID_symbiont.csv")[,c(2,3,5)]

symbs <- symbs %>% 
  mutate(domSymb = ifelse(percent > 50, "C", "D"))

s_df <- s_df %>% 
  left_join(symbs, by = c("coral" = "Sample"))

df_90 <- df_90 %>% 
  left_join(symbs, by = c("coral" = "Sample"))

```


## Principal component analyses {.tabset}

#### *Methods:*
Principal component analysis (PCA) (function *prcomp*) of scaled and centered physiological parameters (host carbohydrate, host lipid, host protein, algal endosymbiont chlorophyll a, algal endosymbiont cell density, holobiont calcification rate as previously for the same samples in Bove et al. (2019)) were employed to assess the relationship between physiological parameters and treatment conditions for each coral species. Main effects (temperature, *p*CO~2~, and reef environment) were evaluated with PERMANOVA using the *adonis2* function (*vegan* package; version `r packageVersion("vegan")`).

<br/>

#### *Results:*
Two principal components (PCs) explained approximately 66% of the variance in physiological responses of the *S. siderea* holobiont to ocean acidification and warming treatments (**Figure 1A**). PC1 was driven by differences in algal endosymbiont physiology (chlorophyll a, cell density), while PC2 represented an inverse relationship between host energy reserves (lipid, protein, carbohydrate) and calcification rates and colour intensities. Overall, lower *p*CO~2~ and temperature resulted in  higher *S. siderea* holobiont physiology (**Figure 1A**). Treatment *p*CO~2~ predominantly drove *S. siderea* physiological responses (p < 0.001; **Table S2**), while temperature and reef environment were not as strong of drivers in physiological responses (p > 0.01 and p > 0.01, respectively; **Table S2**). For *P. strigosa*, 74% of the variance in the holobiont responses to treatments was explained by two PCs (**Figure 1B**). PC1 explained most of the variation of physiological parameters with the exception of host lipid content, which was represented in PC2. Holobiont physiology of *P. strigosa* was clearly reduced under warming and was generally higher in the lower *p*CO~2~ treatments (**Figure 1B**). Treatment temperature (p < 0.001; **Table S2**), *p*CO~2~ (p < 0.01; **Table S2**), and natal reef environment all significantly drove coral holobiont physiology (p < 0.001; **Table S2**). Finally, the first two PCs explained about 59% of the total variance of the *P. astreoides* holobiont response to treatment (**Figure 1C**). Coral holobiont samples separated most clearly along PC1 driven primarily by calcification rate and algal endosymbiont density, while PC2 exhibited an inverse relationship between host total carbohydrate and colour intensity. Overall, lower *p*CO~2~ drove higher *P. astreoides* holobiont physiology, while elevated temperature resulted in greater holobiont physiology (**Figure 1C**). Temperature (p < 0.001; **Table S2**) and *p*CO~2~ (p < 0.001; **Table S2**) drove separations in *P. astreoides* holobiont physiology, while reef environment was nonsignificant (p = 0.82; **Table S2**).

<br/>

### *Siderastrea siderea*

```{r SSID PCA}

# set up the dataframe
s_df <- unique(s_df) # remove any duplicate rows
s_df_l <- gather(s_df, param, value, c(14:17,21:23))
s_df$fpco2 <- factor(s_df$fpco2, levels = c("300", "420", "680", "3300"))
sid_pca_df <- s_df[,c(14:17,21:23)]

# run the adonis
#s_pca_mod <- adonis2(sid_pca_df ~ reef * ftemp * fpco2, data = s_df, method = 'eu', permutations = 1000) # below is the model with non sig interactions removed:
s_pca_mod <- adonis2(sid_pca_df ~ fpco2 + ftemp + reef, data = s_df, method = 'eu', permutations = bootnum)
s_pca_mod # view SSID adonis output

# extract pvalues
s_pval <- s_pca_mod["Pr(>F)"]

# perform principal component analysis (PCA)
s_pca <- prcomp(sid_pca_df, center = TRUE, scale= TRUE)

```

```{r SSID PCA plot, fig.height=8, fig.width=8}

# create labels for p values calculated above
s_pco2_pval <-substitute(italic(P[reef])==p, list(p = format(s_pval[1,1], digits = 2)))
s_temp_pval <-substitute(italic(P[temp])==p, list(p = format(s_pval[2,1], digits = 3)))
s_reef_pval <-substitute(italic(P[pCO[2]])==p, list(p = format(s_pval[3,1], digits = 2)))


# temperature
s_temp_pca <- autoplot(s_pca, data = s_df, 
         colour = "ftemp",
         #shape = "fpco2",
         shape = "ftemp",
         fill = "ftemp",               
         frame = TRUE, 
         frame.type = "t", # displaying ellipses with multivariate t-distributions for small n producing heavier tails
         frame.level = 0.95, # using 95% CI for all ellipses
         frame.alpha = 0.01,
         loadings = TRUE, 
         loadings.colour = "black", 
         loadings.label = TRUE,
         loadings.label.colour = "black",         
         loadings.label.size = 4, 
         loadings.label.hjust = 1.5, 
         loadings.label.vjust = 0.5,
         loadings.label.repel = TRUE) +
  #scale_shape_manual("pCO2", labels = c("pre industrial", "current", "end-of-century", "extreme"), values = c(21, 23, 24, 22)) +
  scale_shape_manual("temperature", labels = c("28C", "31C"), values = c(21, 21)) +
  scale_colour_manual("temperature", labels = c("28C", "31C"), values = c("#4393c3", "#b2182b"))+
  scale_fill_manual("temperature", labels = c("28C", "31C"), values = c("#4393c3", NA))+
  guides(color = guide_legend(keyheight = 0.3, nrow = 4, byrow = TRUE, override.aes = list(linetype = c(0, 0)))) +
  theme(legend.title = element_blank(), legend.background = element_rect(fill = "transparent", colour = NA)) +
  annotate("text", x = -0.30, y = -0.38, label = deparse(s_temp_pval), parse = TRUE, size = 3) +
  ggtitle(expression(paste(bold("A)   "), italic("S. siderea"))))

# pco2
s_pco2_pca <- autoplot(s_pca, data = s_df, 
         colour = "fpco2",
         #shape = "ftemp",
         fill = "fpco2",
         shape = "fpco2",
         frame = TRUE, 
         frame.type = "t", # displaying ellipses with multivariate t-distributions for small n producing heavier tails
         frame.level = 0.95, # using 95% CI for all ellipses
         frame.alpha = 0.01,
         loadings = TRUE, 
         loadings.colour = "black", 
         loadings.label = TRUE,
         loadings.label.colour = "black",         
         loadings.label.size = 4, 
         loadings.label.hjust = 1.5, 
         loadings.label.vjust = 0.5,
         loadings.label.repel = TRUE) +
    #scale_shape_manual("temperature", labels = c("28 C", "31 C"), values = c(19,21)) +
    scale_shape_manual("", labels = c("pre industrial", "current", "end-of-century", "extreme"), values = c(21, 23, 24, 22)) +
    scale_fill_manual("", labels = c("pre industrial", "current", "end-of-century", "extreme"), values = c("#b2abd2", "#5e3c99", "#fdb863", "#e66101"))+
    scale_colour_manual("", labels = c("pre industrial", "current", "end-of-century", "extreme"), values = c("#b2abd2", "#5e3c99", "#fdb863", "#e66101"))+
    guides(color = guide_legend(nrow = 4, override.aes = list(linetype = c(0, 0, 0, 0)))) +
    theme(legend.background = element_rect(fill = "transparent", colour = NA)) +
    annotate("text", x = -0.30, y = -0.38, label = deparse(s_pco2_pval), parse = TRUE, size = 3)

# reef
s_reef_pca <- autoplot(s_pca, data = s_df, 
         colour = "reef",
         fill = "reef",
         frame = TRUE, 
         frame.type = "t", # displaying ellipses with multivariate t-distributions for small n producing heavier tails
         frame.level = 0.95, # using 95% CI for all ellipses
         frame.alpha = 0.01,
         loadings = TRUE, 
         loadings.colour = "black", 
         loadings.label = TRUE,
         loadings.label.colour = "black",         
         loadings.label.size = 4, 
         loadings.label.hjust = -0.6, 
         loadings.label.vjust = 0.5,
         loadings.label.repel = TRUE) +
  scale_colour_manual("reef environment", labels = c("offshore", "inshore"), values = c("#00a08b", "#e9aa2b")) +
  scale_fill_manual("reef environment", labels = c("offshore", "inshore"), values = c("#00a08b", "#e9aa2b")) +
  guides(fill = FALSE, color = guide_legend(keyheight = 0.3, nrow = 2, override.aes = list(linetype = c(0, 0))), shape = guide_legend(nrow = 2, override.aes = list(linetype = c(0, 0)))) +
  theme(legend.title = element_blank(), legend.background = element_rect(fill = "transparent", colour = NA)) +
  annotate("text", x = -0.3, y = -0.38, label = deparse(s_reef_pval), parse = TRUE, size = 3) +
  ggtitle(expression(paste(bold("A)   "), italic("S. siderea"))))

```

<br/>
<br/>


### *Pseudodiploria strigosa*

```{r PSTR PCA}

# set up the dataframe
p_df <- unique(p_df) # remove duplicate rows
p_df_l <- gather(p_df, param, value, c(14:16,21:23))
p_df$fpco2 <- factor(p_df$fpco2, levels = c("300", "420", "680", "3300"))
dip_pca_df <- p_df[,c(14:17,21:23)]

# run the adonis
#p_pca_mod <- adonis2(dip_pca_df ~ reef * ftemp * fpco2, data = p_df, method = 'eu', permutations = 1000) # below is the model with non sig interactions removed:
p_pca_mod <- adonis2(dip_pca_df ~ reef + fpco2 + ftemp, data = p_df, method = 'eu', permutations = bootnum)
p_pca_mod # view PSTR adonis output

# extract pvalues
p_pval <- p_pca_mod["Pr(>F)"]

# perform principal component analysis (PCA)
p_pca <- prcomp(dip_pca_df, center = TRUE, scale= TRUE)

```

```{r PSTR PCA plot, fig.height=8, fig.width=8}

# create labels for p values calculated above
p_reef_pval <-substitute(italic(P[reef])==p, list(p = format(p_pval[1,1], digits = 2)))
p_temp_pval <-substitute(italic(P[temp])==p, list(p = format(p_pval[2,1], digits = 3)))
p_pco2_pval <-substitute(italic(P[pCO[2]])==p, list(p = format(p_pval[3,1], digits = 2)))


# temperature
p_temp_pca <- autoplot(p_pca, data = p_df, 
         colour = "ftemp",
         #shape = "fpco2",
         shape = "ftemp",
         fill = "ftemp",               
         frame = TRUE, 
         frame.type = "t", # displaying ellipses with multivariate t-distributions for small n producing heavier tails
         frame.level = 0.95, # using 95% CI for all ellipses
         frame.alpha = 0.01,
         loadings = TRUE, 
         loadings.colour = "black", 
         loadings.label = TRUE,
         loadings.label.colour = "black",         
         loadings.label.size = 4, 
         loadings.label.hjust = 0.5, 
         loadings.label.vjust = 0.6,
         loadings.label.repel = TRUE) +
  #scale_shape_manual("pCO2", labels = c("pre industrial", "current", "end-of-century", "extreme"), values = c(21, 23, 24, 22)) +
  scale_shape_manual("temperature", labels = c("28C", "31C"), values = c(21, 21)) +
  scale_colour_manual("temperature", labels = c("28C", "31C"), values = c("#4393c3", "#b2182b"))+
  scale_fill_manual("temperature", labels = c("28C", "31C"), values = c("#4393c3", NA))+
  guides(color = guide_legend(keyheight = 0.3, nrow = 4, byrow = TRUE, override.aes = list(linetype = c(0, 0)))) +
  theme(legend.title = element_blank(), legend.background = element_rect(fill = "transparent", colour = NA)) +
  annotate("text", x = -0.13, y = -0.38, label = deparse(p_temp_pval), parse = TRUE, size = 3) +
  ggtitle(expression(paste(bold("B)   "), italic("P. strigosa"))))

# pco2
p_pco2_pca <- autoplot(p_pca, data = p_df, 
         colour = "fpco2",
         fill = "fpco2",
         #shape = "ftemp",
         shape = "fpco2",
         frame = TRUE, 
         frame.type = "t", # displaying ellipses with multivariate t-distributions for small n producing heavier tails
         frame.level = 0.95, # using 95% CI for all ellipses
         frame.alpha = 0.01,
         loadings = TRUE, 
         loadings.colour = "black", 
         loadings.label = TRUE,
         loadings.label.colour = "black",         
         loadings.label.size = 4, 
         loadings.label.hjust = 0.5, 
         loadings.label.vjust = 0.6,
         loadings.label.repel = TRUE) +
    #scale_shape_manual("temperature", labels = c("28 C", "31 C"), values = c(19,21)) +
    scale_shape_manual("", labels = c("pre industrial", "current", "end-of-century", "extreme"), values = c(21, 23, 24, 22)) +
    scale_fill_manual("", labels = c("pre industrial", "current", "end-of-century", "extreme"), values = c("#b2abd2", "#5e3c99", "#fdb863", "#e66101"))+
    scale_colour_manual("", labels = c("pre industrial", "current", "end-of-century", "extreme"), values = c("#b2abd2", "#5e3c99", "#fdb863", "#e66101"))+
    guides(color = guide_legend(nrow = 4, override.aes = list(linetype = c(0, 0, 0, 0)))) +
    theme(legend.background = element_rect(fill = "transparent", colour = NA)) +
    annotate("text", x = -0.28, y = -0.38, label = deparse(p_pco2_pval), parse = TRUE, size = 3)

# reef
p_reef_pca <- autoplot(p_pca, data = p_df, 
         colour = "reef",
         fill = "reef",
         frame = TRUE, 
         frame.type = "t", # displaying ellipses with multivariate t-distributions for small n producing heavier tails
         frame.level = 0.95, # using 95% CI for all ellipses
         frame.alpha = 0.01,
         loadings = TRUE, 
         loadings.colour = "black", 
         loadings.label = TRUE,
         loadings.label.colour = "black",         
         loadings.label.size = 4, 
         loadings.label.hjust = -0.6, 
         loadings.label.vjust = 0.5,
         loadings.label.repel = TRUE) +
  scale_colour_manual("reef environment", labels = c("offshore", "inshore"), values = c("#00a08b", "#e9aa2b")) +
  scale_fill_manual("reef environment", labels = c("offshore", "inshore"), values = c("#00a08b", "#e9aa2b")) +
  guides(fill = FALSE, color = guide_legend(keyheight = 0.3, nrow = 2, override.aes = list(linetype = c(0, 0))), shape = guide_legend(nrow = 2, override.aes = list(linetype = c(0, 0)))) +
  theme(legend.title = element_blank(), legend.background = element_rect(fill = "transparent", colour = NA)) +
  annotate("text", x = -0.17, y = -0.38, label = deparse(p_reef_pval), parse = TRUE, size = 3) +
  ggtitle(expression(paste(bold("B)   "), italic("P. strigosa"))))

```

<br/>
<br/>


### *Porites astreoides*

```{r PAST PCA}

# set up the dataframe
a_df <- unique(a_df)
a_df <- a_df[-17,] # we have two from the same colony in 400_28 that are performing similarly so one is being removed (CFAD12)
a_df_l <- gather(a_df, param, value, c(14:16,21:23))
a_df$fpco2 <- factor(a_df$fpco2, levels = c("300", "420", "680", "3300"))
por_pca_df <- a_df[,c(14:16,18,21:23)]

# run the adonis
#a_pca_mod <- adonis2(por_pca_df ~ fpco2 * ftemp * reef, data = a_df, method = 'eu', permutations = 1000) # below is the model with non sig interactions removed:
a_pca_mod <- adonis2(por_pca_df ~ reef + ftemp + fpco2, data = a_df, method = 'eu', permutations = bootnum)
a_pca_mod # view PAST adonis output

# extract pvalues
a_pval <- a_pca_mod["Pr(>F)"]

# perform principal component analysis (PCA)
a_pca <- prcomp(por_pca_df, center = TRUE, scale= TRUE)

```

```{r PAST PCA plot, fig.height=8, fig.width=8}

# create labels for p values calculated above
a_reef_pval <-substitute(italic(P[reef])==p, list(p = format(a_pval[1,1], digits = 2)))
a_temp_pval <-substitute(italic(P[temp])==p, list(p = format(a_pval[2,1], digits = 3)))
a_pco2_pval <-substitute(italic(P[pCO[2]])==p, list(p = format(a_pval[3,1], digits = 2)))


# temperature
a_temp_pca <- autoplot(a_pca, data = a_df, 
         colour = "ftemp",
         #shape = "fpco2",
         shape = "ftemp",
         fill = "ftemp",               
         frame = TRUE, 
         frame.type = "t", # displaying ellipses with multivariate t-distributions for small n producing heavier tails
         frame.level = 0.95, # using 95% CI for all ellipses
         frame.alpha = 0.01,
         loadings = TRUE, 
         loadings.colour = "black", 
         loadings.label = TRUE,
         loadings.label.colour = "black",         
         loadings.label.size = 4, 
         loadings.label.hjust = -0.6, 
         loadings.label.vjust = 0.5,
         loadings.label.repel = TRUE) +
  #scale_shape_manual("pCO2", labels = c("pre industrial", "current", "end-of-century", "extreme"), values = c(21, 23, 24, 22)) +
  scale_shape_manual("temperature", labels = c("28C", "31C"), values = c(21, 21)) +
  scale_colour_manual("temperature", labels = c("28C", "31C"), values = c("#4393c3", "#b2182b"))+
  scale_fill_manual("temperature", labels = c("28C", "31C"), values = c("#4393c3", NA))+
  guides(color = guide_legend(keyheight = 0.3, nrow = 4, byrow = TRUE, override.aes = list(linetype = c(0, 0)))) +
  theme(legend.title = element_blank(), legend.background = element_rect(fill = "transparent", colour = NA)) +
  annotate("text", x = -0.22, y = -0.44, label = deparse(a_temp_pval), parse = TRUE, size = 3) +
  ggtitle(expression(paste(bold("C)   "), italic("P. astreoides"))))

# pco2
a_pco2_pca <- autoplot(a_pca, data = a_df, 
         colour = "fpco2",
         fill = "fpco2",
         #shape = "ftemp",
         shape = "fpco2",
         frame = TRUE, 
         frame.type = "t", # displaying ellipses with multivariate t-distributions for small n producing heavier tails
         frame.level = 0.95, # using 95% CI for all ellipses
         frame.alpha = 0.01,
         loadings = TRUE, 
         loadings.colour = "black", 
         loadings.label = TRUE,
         loadings.label.colour = "black",         
         loadings.label.size = 4, 
         loadings.label.hjust = -0.6, 
         loadings.label.vjust = 0.5,
         loadings.label.repel = TRUE) +
    #scale_shape_manual("temperature", labels = c("28 C", "31 C"), values = c(19,21)) +
    scale_shape_manual("", labels = c("pre industrial", "current", "end-of-century", "extreme"), values = c(21, 23, 24, 22)) +
    scale_fill_manual("", labels = c("pre industrial", "current", "end-of-century", "extreme"), values = c("#b2abd2", "#5e3c99", "#fdb863", "#e66101"))+
    scale_colour_manual("", labels = c("pre industrial", "current", "end-of-century", "extreme"), values = c("#b2abd2", "#5e3c99", "#fdb863", "#e66101"))+
    guides(color = guide_legend(nrow = 4, override.aes = list(linetype = c(0, 0, 0, 0)))) +
    theme(legend.background = element_rect(fill = "transparent", colour = NA)) +
  annotate("text", x = -0.167, y = -0.5, label = deparse(a_pco2_pval), parse = TRUE, size = 3)

# reef
a_reef_pca <- autoplot(a_pca, data = a_df, 
         colour = "reef",
         fill = "reef",
         frame = TRUE, 
         frame.type = "t", # displaying ellipses with multivariate t-distributions for small n producing heavier tails
         frame.level = 0.95, # using 95% CI for all ellipses
         frame.alpha = 0.01,
         loadings = TRUE, 
         loadings.colour = "black", 
         loadings.label = TRUE,
         loadings.label.colour = "black",         
         loadings.label.size = 4, 
         loadings.label.hjust = -0.6, 
         loadings.label.vjust = 0.5,
         loadings.label.repel = TRUE) +
  scale_colour_manual("reef environment", labels = c("offshore", "inshore"), values = c("#00a08b", "#e9aa2b")) +
  scale_fill_manual("reef environment", labels = c("offshore", "inshore"), values = c("#00a08b", "#e9aa2b")) +
  guides(fill = FALSE, color = guide_legend(keyheight = 0.3, nrow = 2, override.aes = list(linetype = c(0, 0))), shape = guide_legend(nrow = 2, override.aes = list(linetype = c(0, 0)))) +
  theme(legend.title = element_blank(), legend.background = element_rect(fill = "transparent", colour = NA)) +
  annotate("text", x = -0.245, y = -0.38, label = deparse(a_reef_pval), parse = TRUE, size = 3) +
  ggtitle(expression(paste(bold("C)   "), italic("P. astreoides"))))

```

<br/>
<br/>


### **Figure 1**

```{r PCA plot, fig.height=7, fig.width=13}

temp_PCA <- ggarrange(s_temp_pca, p_temp_pca, a_temp_pca, ncol = 3, common.legend = TRUE, legend = "right")
pCO2_PCA <- ggarrange(s_pco2_pca, p_pco2_pca, a_pco2_pca, ncol = 3, common.legend = TRUE, legend = "right")

ggarrange(temp_PCA, pCO2_PCA, nrow = 2) +
  ggsave("Figures/Final_Figures/Figure1_PhysPCA.pdf", width = 13, height = 7) +
  ggsave("Figures/Final_Figures/Figure1_PhysPCA.png", width = 13, height = 7)


```

**Figure 1.** Principal component analysis (PCA) of all coral holobiont physiological parameters for *S. siderea* (**A**), *P. strigosa* (**B**), and *P. astreoides* (**C**) after 93 days of exposure to different temperature and *p*CO~2~ treatments. PCAs in the top row are depicted by temperature treatment for each species (28$^\circ$ C blue; 31$^\circ$ C red) and the bottom row of PCAs are depicted by *p*CO~2~ for each species (280 $\mu$atm light purple; 400 $\mu$atm dark purple; 700 $\mu$atm light orange; 2800 $\mu$atm dark orange). Arrows represent significant (p < 0.05) correlation vectors for physiological parameters and ellipses represent 95% confidence based on multivariate t-distributions.

<br/>
<br/>


### *Figure S1*

```{r PCA plot reefs, fig.height = 3.8, fig.width = 13}

ggarrange(s_reef_pca, p_reef_pca, a_reef_pca, ncol = 3, common.legend = TRUE, legend = "right") +
  ggsave("Figures/Supplemental_Figures/FigureS1_PhysPCA_RZ.pdf", width = 13, height = 3.8) +
  ggsave("Figures/Supplemental_Figures/FigureS1_PhysPCA_RZ.png", width = 13, height = 3.8)

```

**Figure S1.** Principal component analysis (PCA) of all coral holobiont physiological parameters for *S. siderea* (**A**), *P. strigosa* (**B**), and *P. astreoides* (**C**) depicted by natal reef environment (offshore green; inshore yellow). Arrows represent significant (p < 0.05) correlation vectors for physiological parameters and ellipses represent 95% confidence based on multivariate t-distributions.

<br/>
<br/>


## Correlation assessments {.tabset}

#### *Methods:*
Correlations of all physiological parameters were assessed to determine the relationships between parameters within each species, redargless of temperature and *p*CO~2~ treatment. The Pearson correlation coefficient (R^2^) of each comparison was calculated using the corrgram package (version `r packageVersion("corrgram")`) and the significance was calculated using the *cor.test* function. These relationships were then visualized through simple scatterplots.

<br/>

#### *Results:*
Correlations of coral holobiont physiological parameters were generally positively related with one another across all three species. Correlations between *S. siderea* holobiont physiological parameters identified 15 significant relationships out of all 21 possible comparisons (**Figure 2A**). Of those significant correlations, six resulted in a Pearson's correlation coefficient (R^2^) equal to or greater than 0.5, with the strongest relationship identified being symbiont density vs chlorophyll a (R^2^ = 0.72). All pairwise physiological parameters were significantly correlated with one another in *P. strigosa* and of those, 14 correlations exhibit moderate (R^2^ > 0.50) positive relationships (**Figure 2B**). Notably, the two strongest correlations were host carbohydrate vs host protein (R^2^ = 0.70) and host carbohydrate vs chlorophyll a (R^2^ = 0.76). Compared to both *S. siderea* and *P. strigosa*, fewer physiological traits were significantly (p < 0.05) correlated with one another in *P. astreoides* (12 significant out of 21 total comparisons; **Figure 2C**). Of the significant correlations, only two pairwise comparisons resulted in a Pearson's correlation coefficient greater than 0.5: chlorophyll a vs colour intensity (R^2^ = 0.57) and host carbohydrate vs host protein (R^2^ = 0.68).

<br/>

```{r functions for custom correlation and scatter plots}

## Custom function for the border around the text
panel.txt.brdr <- function (x = 0.5, y = 0.5, txt, cex, font, srt) 
{
  text(x, y, txt, cex = cex, font = font, srt = srt)
  box(col = "#525252")
}

## Custom function for the top point fill/shape panel
panel.pts.col <- function (x, y, corr = NULL, col.regions, cor.method, color, ...) 
{
  if (!is.null(corr)) 
    return()
  plot.xy(xy.coords(x, y), col = color, type = "p", ...)
  box(col = "#525252")
}

## Custom function for the bottom fill/R2 panel (with significance)
panel.fill.R2 <- function (x, y, corr = NULL, col.regions, cor.method, digits = 2, cex.cor, ...) 
{
  if (is.null(corr)) {
    if (sum(complete.cases(x, y)) < 2) {
      warning("Need at least 2 complete cases for cor()")
      return()
    }
    else {
      corr <- cor(x, y, use = "pair", method = cor.method)
    }
  }
  ncol <- 14
  pal <- col.regions(ncol)
  col.ind <- as.numeric(cut(corr, breaks = seq(from = -1, 
                                               to = 1, length.out = ncol + 1), include.lowest = TRUE))
  usr <- par("usr")
  rect(usr[1], usr[3], usr[2], usr[4], col = pal[col.ind], 
       border = NA)
  box(col = "#525252")
  
  auto <- missing(cex.cor)
  on.exit(par(usr))
  par(usr = c(0, 1, 0, 1))
  abscorr <- formatC(abs(corr), digits = digits, format = "f")
  corr2 <- formatC(corr, digits = digits, format = "f")
  pval <- cor.test(x, y, use = "pair", method = cor.method)[["p.value"]]
  sig <- if(pval > 0.05){""}else if(pval < 0.05 & pval > 0.01){"*"}else if(pval < 0.01 & pval > 0.001){"**"}else if(pval < 0.001){"***"}
  if (auto) 
    cex.cor <- 0.4/strwidth(abscorr)
  text(0.5, 0.5, paste0(corr2, sig), cex = cex.cor, font = 3, col = "#252525")
  
  
  
}

```

```{r SSID correlation matrix and scatter plots, results='hide'}

## Create new correlation dataframe for SSID
sid_corr_df <- s_df[,c(10:11,14:17,21:23)] %>% 
  mutate(shape = case_when(ftemp == "28" ~ "19",
                           ftemp == "31" ~ "21"),
         colour = case_when(fpco2 == "300" ~ "#b2abd2",
                            fpco2 == "420" ~ "#5e3c99",
                            fpco2 == "680" ~ "#fdb863",
                            fpco2 == "3300" ~ "#e66101"))

## Plot correlation matrix and scatter plot of all SSID physiology  
png(file = "Figures/Supplemental_Figures/SSID_PhysCorrelations.png", width = 330, height = 225, units='mm', res = 300)  

corrgram(sid_corr_df, order = FALSE, lower.panel = panel.fill.R2, upper.panel = panel.pts.col, text.panel = panel.txt.brdr,
         col.regions = colorRampPalette(c("#ffffcc", "#c7e9b4", "#225ea8")), 
         color = sid_corr_df$colour, pch = as.numeric(as.character(sid_corr_df$shape)),
         labels = c("symbiont \ndensity", "host \nprotein", "chlorophyll a", "colour \nintensity", "calcification \nrate", "host \ncarbohydrate", "host \nlipid")) 

dev.off() 

```

```{r PSTR correlation matrix and scatter plots, results='hide'}

## Create new correlation dataframe for SSID
dip_corr_df <- p_df[,c(10:11,14:17,21:23)] %>% 
  mutate(shape = case_when(ftemp == "28" ~ "19",
                           ftemp == "31" ~ "21"),
         colour = case_when(fpco2 == "300" ~ "#b2abd2",
                            fpco2 == "420" ~ "#5e3c99",
                            fpco2 == "680" ~ "#fdb863",
                            fpco2 == "3300" ~ "#e66101"))
  

## Plot correlation matrix and scatter plot of all PSTR physiology
png(file = "Figures/Supplemental_Figures/PSTR_PhysCorrelations.png", width = 330, height = 225, units='mm', res = 300) 

corrgram(dip_corr_df, order = FALSE, lower.panel=panel.fill.R2, upper.panel=panel.pts.col, text.panel=panel.txt.brdr,
         col.regions=colorRampPalette(c("#ffffcc", "#c7e9b4", "#225ea8")), 
         color = dip_corr_df$colour, pch = as.numeric(as.character(dip_corr_df$shape)),
         labels = c("symbiont \ndensity", "host \nprotein", "chlorophyll a", "colour \nintensity", "calcification \nrate", "host \ncarbohydrate", "host \nlipid")) 

dev.off() 

```

```{r PAST correlation matrix and scatter plots, results='hide'}

## Create new correlation dataframe for SSID
por_corr_df <- a_df[,c(10:11,14:16,18,21:23)] %>% 
  mutate(shape = case_when(ftemp == "28" ~ "19",
                           ftemp == "31" ~ "21"),
         colour = case_when(fpco2 == "300" ~ "#b2abd2",
                            fpco2 == "420" ~ "#5e3c99",
                            fpco2 == "680" ~ "#fdb863",
                            fpco2 == "3300" ~ "#e66101"))
  

## Plot correlation matrix and scatter plot of all PAST physiology
png(file = "Figures/Supplemental_Figures/PAST_PhysCorrelations.png", width = 330, height = 225, units='mm', res = 300) 

corrgram(por_corr_df, order = FALSE, lower.panel=panel.fill.R2, upper.panel=panel.pts.col, text.panel=panel.txt.brdr,
         col.regions=colorRampPalette(c("#ffffcc", "#c7e9b4", "#225ea8")), 
         color = por_corr_df$colour, pch = as.numeric(as.character(por_corr_df$shape)),
         labels = c("symbiont \ndensity", "host \nprotein", "chlorophyll a", "colour \nintensity", "calcification \nrate", "host \ncarbohydrate", "host \nlipid")) 

dev.off() 

```

```{r save all species correlation/scatter plots, fig.width = 8, fig.height = 6.4}

ssid_corr_plot <- readPNG("Figures/Supplemental_Figures/SSID_PhysCorrelations.png")
pstr_corr_plot <- readPNG("Figures/Supplemental_Figures/PSTR_PhysCorrelations.png")
past_corr_plot <- readPNG("Figures/Supplemental_Figures/PAST_PhysCorrelations.png")

ssid_corr_plot <- ggplot() + 
  background_image(ssid_corr_plot) + 
  theme_void() + 
  ggtitle(expression(paste(bold("  A)   "), italic("S. siderea"))))

pstr_corr_plot <- ggplot() + 
  background_image(pstr_corr_plot) + 
  theme_void() + 
  ggtitle(expression(paste(bold("  B)   "), italic("P. strigosa"))))

past_corr_plot <- ggplot() + 
  background_image(past_corr_plot) + 
  theme_void() + 
  ggtitle(expression(paste(bold("  B)   "), italic("P. asteroides"))))

ggarrange(ssid_corr_plot, pstr_corr_plot, past_corr_plot, nrows = 2) +
  ggsave("Figures/Final_Figures/Figure2_PhysCorrelations.pdf", width = 8, height = 6.4) +
  ggsave("Figures/Final_Figures/Figure2_PhysCorrelations.png", width = 8, height = 6.4)

```

**Figure 2.** Coral holobiont correlation matrices (bottom panel) and scatter plots (top panel) for *S. siderea* (**A**), *P. strigosa* (**B**), and *P. astreoides* (**C**) depicting pairwise comparisons of physiological parameters within each species. Strength of correlations between parameters is indicated by darker shades of blue in the bottom panel with a higher R^2^ value (Pearson correlation coefficient). Of these correlations, significant correlations are depicted with asterisks according to significance level (\* p < 0.05; \*\* p < 0.01; \*\*\* p < 0.001). Scatter plots of physiological parameters are displayed in the top panel with temperature depicted by shape (28$^\circ$C filled points; 31$^\circ$C open points) and *p*CO~2~ depicted by colour (280 $\mu$atm light purple; 400 $\mu$atm dark purple; 700 $\mu$atm light orange; 2800 $\mu$atm dark orange).

<br/>
<br/>


## Plasticity analyses {.tabset} 

#### *Methods:*
Using PC1 and PC2 for each species, we then calculated the phenotypic plasticity of each experimental fragment. Plasticity was calculated as the PC distance between an experimental fragment and the control (400 $\mu$atm; 28$^\circ$C) fragment from that same colony. The effects of treatment (*p*CO~2~ and temperature) and natal reef environment on calculated distances were assessed using generalized linear models (function *glm*) with a Gamma distribution and log-link. The best-fit model was selected as the model with the lowest AIC for each species (**Table S1**). The main effects of treatment and reef environment were assessed with an ANOVA (package *car*; version `r packageVersion("car")`) with a type III error. 

<br/>

#### *Results:*
Natal reef environment (p < 0.05) and *p*CO~2~ (p < 0.05) significantly altered the phenotypic plasticity of *S. siderea* (**Figure 3A; Table S3, S4**). Offshore fragments exhibited a positive linear trend with increasing *p*CO~2~ while the inshore fragments appear to respond in a parabolic pattern to *p*CO~2~, with the lowest calculated distances occurring at 400 $\mu$atm, 31$^\circ$C and 700 $\mu$atm, 28$^\circ$C. Plasticity of *P. strigosa* and *P. astreoides* was not significantly altered by temperature treatment, *p*CO~2~ treatment, or natal reef environment (**Figure 3B, 3C; Table S3, S4**). However, *P. astreoides* exhibited a slight trend in the inshore fragments suggesting potentially higher plasticity with increasing *p*CO~2~ that is not seen in the offshore fragments (**Figure 3C**).

<br/>

### *Siderastrea siderea*

```{r SSID distance dataframe setup}

s_dis <- s_pca$x[,1:2] # grab PC1 and PC2 distances from prcomp() object
sid_dist <- cbind(s_df[,c(1,7,10,11,12,27)], s_dis) # combine the datasets

# adds column with the control value PC1/PC2 per colony
sid_dist$cont_pc1[sid_dist$treat2 == "447_28"] <- sid_dist$PC1[sid_dist$treat2 == "447_28"]
sid_dist$cont_pc2[sid_dist$treat2 == "447_28"] <- sid_dist$PC2[sid_dist$treat2 == "447_28"]

# get the 400_28 PC values
sid_dist2 <- sid_dist %>%  
  group_by(colony) %>% 
  summarise(con2_pc1 = sum(cont_pc1, na.rm = TRUE),
            con2_pc2 = sum(cont_pc2, na.rm = TRUE))

# Make columns of the 400_28 PC values
sid_dist <- sid_dist %>% 
  left_join(sid_dist2, by = c("colony" = "colony"))

sid_dist$dist <- sqrt(((sid_dist$PC1 - sid_dist$con2_pc1)^2) + ((sid_dist$PC2 - sid_dist$con2_pc2)^2)) # formula for calculating distance between control (T90) and treatment values
sid_dist$treat_plot = paste(sid_dist$fpco2, sid_dist$ftemp, sep = "_")


## modify dataframe to remove the control treatment and reorder levels
sid_dist <- sid_dist %>% 
  filter(treat_plot != "420_28") %>% 
  mutate(treat_plot = factor(treat_plot, levels = c("300_28", "300_31", "420_28", "420_31", "680_28", "680_31", "3300_28", "3300_31")),
         reef = factor(reef, levels = c("N", "F")),
         ftemp = factor(ftemp, levels = c("28", "31")),
         fpco2 = factor(fpco2, levels = c("300", "420", "680", "3300")))

```

```{r SSID distance analysis, include=FALSE}

## Model selection (via AIC)
ssid_dist_mod <- glm(dist ~ reef * fpco2 * ftemp, family = Gamma(link = "log"), data = sid_dist)
ssid_dist_mod2 <- glm(dist ~ reef * fpco2 + ftemp, family = Gamma(link = "log"), data = sid_dist) # best fit
ssid_dist_mod3 <- glm(dist ~ reef + fpco2 * ftemp, family = Gamma(link = "log"), data = sid_dist)
ssid_dist_mod4 <- glm(dist ~ reef + fpco2 + ftemp, family = Gamma(link = "log"), data = sid_dist)
ssid_dist_mod5 <- glm(dist ~ reef * (fpco2 + ftemp), family = Gamma(link = "log"), data = sid_dist)
ssid_dist_mod6 <- glm(dist ~ fpco2 + ftemp, family = Gamma(link = "log"), data = sid_dist)

# check for best-fit model
ssid_plast_aic <- compare_performance(ssid_dist_mod, ssid_dist_mod2, ssid_dist_mod3, ssid_dist_mod4, ssid_dist_mod5, ssid_dist_mod6, rank = TRUE)
plot(ssid_plast_aic)


## Best-fit GLM with Gamma log link
ssid_dist_glm <- glm(dist ~ reef * fpco2 + ftemp, family = Gamma(link = "log"), data = sid_dist, contrasts = list(reef=contr.sum, fpco2=contr.sum, ftemp=contr.sum))
ssid_glm_out <- summary(ssid_dist_glm) # summary output of the GLM
ssid_anova_out <- Anova(ssid_dist_glm, type = "III", test.statistic = "F") # comparison of the 'main effects'





#### Can't seem to make bootstrapping work for PSTR so just going to report mean/SE in the plots and then the GLM model output
### Bootstrapping with boot_predict() function in package finalfit (version 1.0.2)
#newdata_ssid <- data.frame(reef = sid_dist$reef, fpco2 = sid_dist$fpco2, ftemp = sid_dist$ftemp, treat_plot = sid_dist$treat_plot, dist = sid_dist$dist)
#
#ssid_dist_boot <- boot_predict(ssid_dist_glm, newdata = newdata_ssid, R = bootnum, boot_compare = FALSE,  estimate_name = "Predicted plasticity") %>% # compute boot predictions
#  mutate(plasticity = as.character(`Predicted plasticity`), # create second predicted column for plotting
#         plasticity = gsub("[()to]", "", plasticity)) %>% # remove characters
#  separate(plasticity, c("estimate", "lowerci", "rm", "upperci"), " ") %>% # separate into three columns for estimate, lower CI, and upper CI
#  dplyr::select(-rm) %>% # remove the blank column created
#  mutate_if(is.character, as.numeric) # convert the esimate/CI from character to numeric
  
```

```{r plots of SSID distances, fig.height = 3.5, fig.width = 6}

## Summary data for plotting based on mean/SE instead of boot
sid_dist_sum <- sid_dist %>% 
  mutate(treat_plot = factor(treat_plot, levels = c("300_28", "300_31", "420_28", "420_31", "680_28", "680_31", "3300_28", "3300_31"))) %>% 
  group_by(treat_plot, reef) %>% 
  summarise(mean = mean(dist),
            se = (sd(dist) / sqrt(n()))) 


## Plasticity plot
ssid_plast_plot <- ggplot(sid_dist_sum, aes(x = reef, y = mean, colour = treat_plot, fill = treat_plot, shape = treat_plot)) +
  theme_pubr() +
  theme(legend.title = element_blank(), axis.ticks.x = element_blank()) +
  guides(shape = guide_legend(ncol = 1), fill = guide_legend(override.aes = list(linetype = 0)), color = guide_legend(override.aes = list(linetype = 0))) +
  geom_point(data = sid_dist, aes(x = reef, y = dist), size = 2, alpha = 0.7, position = position_dodge(width = 0.55)) +
  geom_errorbar(aes(ymin = mean - se, ymax = mean + se), alpha = 0.8, size = 1, width = 0, position = position_dodge(width = 0.55)) +
  geom_point(size = 3, stroke = 1, position = position_dodge(width = 0.55)) +
  scale_shape_manual(values = c(21, 21, 23, 24, 24, 22, 22)) +
  scale_colour_manual(values = c("#b2abd2", "#b2abd2", "#5e3c99", "#fdb863", "#fdb863", "#e66101", "#e66101")) + 
  scale_fill_manual(values = c("#b2abd2", "white", "white", "#fdb863", "white", "#e66101", "white")) + 
  labs(y = "PC distance from control", x = "") +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 8)) +
  scale_x_discrete(labels = c("F" = "Offshore", "N" = "Inshore"))
  

## Model output
ssid_glm_out # summary output of the GLM
ssid_anova_out # comparison of the 'main effects'

```

<br/>
<br/>


### *Pseudodiploria strigosa*

```{r PSTR distance dataframe setup}

p_dis <- p_pca$x[,1:2] # grab PC1 and PC2 distances from prcomp() object
dip_dist <- cbind(p_df[,c(1,7,10,11,12,27)], p_dis) # combine the datasets

# adds column with the control value PC1/PC2 per colony
dip_dist$cont_pc1[dip_dist$treat2 == "447_28"] <- dip_dist$PC1[dip_dist$treat2 == "447_28"]
dip_dist$cont_pc2[dip_dist$treat2 == "447_28"] <- dip_dist$PC2[dip_dist$treat2 == "447_28"]

# get the 400_28 PC values
dip_dist2 <- dip_dist %>%  
  group_by(colony) %>% 
  summarise(con2_pc1 = sum(cont_pc1, na.rm = TRUE),
            con2_pc2 = sum(cont_pc2, na.rm = TRUE))

# Make columns of the 400_28 PC values
dip_dist <- dip_dist %>% 
  left_join(dip_dist2, by = c("colony" = "colony"))

dip_dist$dist <- sqrt(((dip_dist$PC1 - dip_dist$con2_pc1)^2) + ((dip_dist$PC2 - dip_dist$con2_pc2)^2)) # formula for 
dip_dist$treat_plot = paste(dip_dist$fpco2, dip_dist$ftemp, sep = "_")


## modify dataframe to remove the control treatment and reorder levels
dip_dist <- dip_dist %>% 
  filter(treat_plot != "420_28")%>% 
  mutate(treat_plot = factor(treat_plot, levels = c("300_28", "300_31", "420_28", "420_31", "680_28", "680_31", "3300_28", "3300_31")),
         reef = factor(reef, levels = c("N", "F")),
         ftemp = factor(ftemp, levels = c("28", "31")),
         fpco2 = factor(fpco2, levels = c("300", "420", "680", "3300")))

```

```{r PSTR distance analysis, include=FALSE}

## Model selection (via AIC)
pstr_dist_mod <- glm(dist ~ reef * fpco2 * ftemp, family = Gamma(link = "log"), data = dip_dist)
pstr_dist_mod2 <- glm(dist ~ reef * fpco2 + ftemp, family = Gamma(link = "log"), data = dip_dist)
pstr_dist_mod3 <- glm(dist ~ reef + fpco2 * ftemp, family = Gamma(link = "log"), data = dip_dist) # best fit but results in singularity for 400_31 so moving forward with mod4
pstr_dist_mod4 <- glm(dist ~ reef + fpco2 + ftemp, family = Gamma(link = "log"), data = dip_dist) # using this model
pstr_dist_mod5 <- glm(dist ~ reef * (fpco2 + ftemp), family = Gamma(link = "log"), data = dip_dist)
pstr_dist_mod6 <- glm(dist ~ fpco2 + ftemp, family = Gamma(link = "log"), data = dip_dist)

# check for best-fit model
pstr_plast_aic <- compare_performance(pstr_dist_mod, pstr_dist_mod2, pstr_dist_mod3, pstr_dist_mod4, pstr_dist_mod5, pstr_dist_mod6, rank = TRUE)
plot(pstr_plast_aic)


## Best-fit GLM with Gamma log link
pstr_dist_glm <- glm(dist ~ reef + fpco2 + ftemp, family = Gamma(link = "log"), data = dip_dist, contrasts = list(reef=contr.sum, fpco2=contr.sum, ftemp=contr.sum))
pstr_glm_out <- summary(pstr_dist_glm) # summary output of the GLM
pstr_anova_out <- Anova(pstr_dist_glm, type = "III", test.statistic = "F") # comparison of the 'main effects'

```
  
```{r plots of PSTR distances, fig.height = 3.5, fig.width = 6}

## Summary data for plotting
dip_dist_sum <- dip_dist %>% 
  mutate(treat_plot = factor(treat_plot, levels = c("300_28", "300_31", "420_28", "420_31", "680_28", "680_31", "3300_28", "3300_31"))) %>% 
  group_by(treat_plot, reef) %>% 
  summarise(mean = mean(dist),
            se = (sd(dist) / sqrt(n()))) 


## Plasticity plot
pstr_plast_plot <- ggplot(dip_dist_sum, aes(x = reef, y = mean, colour = treat_plot, fill = treat_plot, shape = treat_plot)) +
  theme_pubr() +
  theme(legend.title = element_blank(), axis.ticks.x = element_blank()) +
  guides(shape = guide_legend(ncol = 1), fill = guide_legend(override.aes = list(linetype = 0)), color = guide_legend(override.aes = list(linetype = 0))) +
  geom_point(data = dip_dist, aes(x = reef, y = dist), size = 2, alpha = 0.7, position = position_dodge(width = 0.55)) +
  geom_errorbar(aes(ymin = mean - se, ymax = mean + se), alpha = 0.8, size = 1, width = 0, position = position_dodge(width = 0.55)) +
  geom_point(size = 3, stroke = 1, position = position_dodge(width = 0.55)) +
  scale_shape_manual(values = c(21, 21, 23, 24, 24, 22, 22)) +
  scale_colour_manual(values = c("#b2abd2", "#b2abd2", "#5e3c99", "#fdb863", "#fdb863", "#e66101", "#e66101")) + 
  scale_fill_manual(values = c("#b2abd2", "white", "white", "#fdb863", "white", "#e66101", "white")) + 
  labs(y = "", x = "") +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 8)) +
  scale_x_discrete(labels = c("F" = "Offshore", "N" = "Inshore"))


## Model output
pstr_glm_out # summary output of the GLM
pstr_anova_out # comparison of the 'main effects'

```

<br/>
<br/>


### *Porites astreoides*

```{r PAST distance dataframe setup}

a_dis <- a_pca$x[,1:2] # grab PC1 and PC2 distances from prcomp() object
por_dist <- cbind(a_df[,c(1,7,10,11,12,27)], a_dis) # combine the datasets

# adds column with the control value PC1/PC2 per colony
por_dist$cont_pc1[por_dist$treat2 == "447_28"] <- por_dist$PC1[por_dist$treat2 == "447_28"]
por_dist$cont_pc2[por_dist$treat2 == "447_28"] <- por_dist$PC2[por_dist$treat2 == "447_28"]

# get the 400_28 PC values
por_dist2 <- por_dist %>%  
  group_by(colony) %>% 
  summarise(con2_pc1 = sum(cont_pc1, na.rm = TRUE),
            con2_pc2 = sum(cont_pc2, na.rm = TRUE))

# Make columns of the 400_28 PC values
por_dist <- por_dist %>% 
  left_join(por_dist2, by = c("colony" = "colony"))

por_dist$dist <- sqrt(((por_dist$PC1 - por_dist$con2_pc1)^2) + ((por_dist$PC2 - por_dist$con2_pc2)^2)) # formula for calculating distance between control (T90) and treatment values
por_dist$treat_plot = paste(por_dist$fpco2, por_dist$ftemp, sep = "_")

## modify dataframe to remove the control treatment and reorder levels
por_dist <- por_dist %>% 
  filter(treat_plot != "420_28")%>% 
  mutate(treat_plot = factor(treat_plot, levels = c("300_28", "300_31", "420_28", "420_31", "680_28", "680_31", "3300_28", "3300_31")),
         reef = factor(reef, levels = c("N", "F")),
         ftemp = factor(ftemp, levels = c("28", "31")),
         fpco2 = factor(fpco2, levels = c("300", "420", "680", "3300")))

```

```{r PAST distance analysis, include=FALSE}

## Model selection (via AIC)
past_dist_mod <- glm(dist ~ reef * fpco2 * ftemp, family = Gamma(link = "log"), data = por_dist)
past_dist_mod2 <- glm(dist ~ reef * fpco2 + ftemp, family = Gamma(link = "log"), data = por_dist)
past_dist_mod3 <- glm(dist ~ reef + fpco2 * ftemp, family = Gamma(link = "log"), data = por_dist)
past_dist_mod4 <- glm(dist ~ reef + fpco2 + ftemp, family = Gamma(link = "log"), data = por_dist) # using this model
past_dist_mod5 <- glm(dist ~ reef * (fpco2 + ftemp), family = Gamma(link = "log"), data = por_dist)
past_dist_mod6 <- glm(dist ~ fpco2 + ftemp, family = Gamma(link = "log"), data = por_dist) # best fit but using mod4 to include reef

# check for best-fit model
past_plast_aic <- compare_performance(past_dist_mod, past_dist_mod2, past_dist_mod3, past_dist_mod4, past_dist_mod5, past_dist_mod6, rank = TRUE)
plot(past_plast_aic)


## Best-fit GLM with Gamma log link
past_dist_glm <- glm(dist ~ reef + fpco2 + ftemp, family = Gamma(link = "log"), data = por_dist, contrasts = list(reef=contr.sum, fpco2=contr.sum, ftemp=contr.sum))
past_glm_out <- summary(past_dist_glm) # summary output of the GLM
past_anova_out <- Anova(past_dist_glm, type = "III", test.statistic = "F") # comparison of the 'main effects'

```
  
```{r plots of PAST distances, fig.height = 3, fig.width = 10}

## Summary data for plotting
por_dist_sum <- por_dist %>% 
  mutate(treat_plot = factor(treat_plot, levels = c("300_28", "300_31", "420_28", "420_31", "680_28", "680_31", "3300_28", "3300_31"))) %>% 
  group_by(treat_plot, reef) %>% 
  summarise(mean = mean(dist),
            se = (sd(dist) / sqrt(n()))) 


## Plasticity plot
past_plast_plot <- ggplot(por_dist_sum, aes(x = reef, y = mean, colour = treat_plot, fill = treat_plot, shape = treat_plot)) +
  theme_pubr() +
  theme(legend.title = element_blank(), axis.ticks.x = element_blank()) +
  guides(shape = guide_legend(ncol = 1), fill = guide_legend(override.aes = list(linetype = 0)), color = guide_legend(override.aes = list(linetype = 0))) +
  geom_point(data = por_dist, aes(x = reef, y = dist), size = 2, alpha = 0.7, position = position_dodge(width = 0.55)) +
  geom_errorbar(aes(ymin = mean - se, ymax = mean + se), alpha = 0.8, size = 1, width = 0, position = position_dodge(width = 0.55)) +
  geom_point(size = 3, stroke = 1, position = position_dodge(width = 0.55)) +
  scale_shape_manual(values = c(21, 21, 23, 24, 24, 22, 22)) +
  scale_colour_manual(values = c("#b2abd2", "#b2abd2", "#5e3c99", "#fdb863", "#fdb863", "#e66101", "#e66101")) + 
  scale_fill_manual(values = c("#b2abd2", "white", "white", "#fdb863", "white", "#e66101", "white")) + 
  labs(y = "", x = "") +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 8)) +
  scale_x_discrete(labels = c("F" = "Offshore", "N" = "Inshore"))


## Model output
past_glm_out # summary output of the GLM
past_anova_out # comparison of the 'main effects'

```

<br/>
<br/>


### **Figure 3**

```{r plot and save all species plasticity, fig.width = 13, fig.height = 4}

ggarrange(ssid_plast_plot, pstr_plast_plot, past_plast_plot, common.legend = TRUE, legend = "right", ncol = 3) +
  ggsave("Figures/Final_Figures/Figure3_PhysPlasticity.pdf", width = 13, height = 4) +
  ggsave("Figures/Final_Figures/Figure3_PhysPlasticity.png", width = 13, height = 4)

```

**Figure 3.** Assessment of phenotypic plasticity of *S. siderea* (**A**), *P. strigosa* (**B**), and *P. astreoides* (**C**) in experimental treatments and by natal reef environment. Higher values represent greater plasticity in coral holobiont samples. *p*CO~2~ treatment is depicted by colour and shape (280 $\mu$atm light purple, circle; 400 $\mu$atm dark purple, diamond; 700 $\mu$atm light orange, triangle; 2800 $\mu$atm dark orange, square) and temperature is represented as either filled (28$^\circ$C) or open (31$^\circ$C) symbols.

<br/>
<br/>


## *Siderastrea siderea* subset {.tabset} 

#### *Methods:*
Using the subset of *S. siderea* fragments with known Symbiodinacaea community composition identified from transcriptomic analyses (described below), we further assessed the differences in coral holobiont physiological responses by dominant symbiont genera. Relative abundance of Symbiodinacaea genus within each fragment was assigned as either *Cladocopium spp.* dominated (> 50% reads mapping to *Cladocopium*) or *Durusdinium spp.* dominated (> 50% reads mapping to *Durusdinium*). A PCA was performed with all coral holobiont physiological parameters with the addition of percent of reads mapping to *Cladocopium spp.* such that higher percentages represented *Cladocopium* dominated fragments and lower percentages represented *Durusdinium* dominated fragments. The phenotypic plasticity of the subset of *S. siderea* fragments was also assessed using the same methods described above. The effects of dominant Symbiodinacaea genus and treatment (*p*CO~2~ and temperature) were assessed using a generalized linear models (function *glm*) with a Gamma distribution and log-link and main effects were assessed with an ANOVA (package *car*; version `r packageVersion("car")`) with a type III error. All figures and statistical analyses were carried out in `r paste0("R version ", R.Version()$major, ".", R.Version()$minor)` (R Core Development Team 2016).

<br/>

#### *Results:*
Two principal components (PCs) explained approximately 60% of the variance in holobiont physiological responses of the subset of *S. siderea* fragments (**Figure 4A**). Interestingly, PC2 was driven by differences in Symbiodinacaea genera, along with host lipid content and colour intensity, while PC1 represented host carbohydrate and symbiont cell density. Fragments of *S. siderea* hosting higher abundances of *Cladocopium spp.* symbionts exhibited faster calcification rates while fragments dominated by *Durusdinium spp.* tended to have higher host lipid and protein content (**Figure 4A**). Experimental *p*CO~2~ (p < 0.001) and temperature (p < 0.01), dominant Symbiodinacaea genus (p < 0.001), and natal reef environment (p < 0.01) all significantly altered the holobiont physiology in the subset of *S. siderea* (**Figure S2; Table S5**). Analysis of the subset of *S. siderea* holobiont fragments suggests that dominant Symbiodinacaea genus (p = 0.05) and *p*CO~2~ (p = 0.05) altered the phenotypic plasticity in these samples (**Figure 4B; Table S6, S7**), however, the small sample size and marginal p value should be noted with interpretation.

<br/>

### Principal component analysis

```{r SSID subset PCA}

# set up the subset dataframe
s_df_sub <- s_df[!is.na(s_df$percent), ]
s_df_sub$fpco2 <- factor(s_df_sub$fpco2, levels = c("300", "420", "680", "3300"))
sid_sub_pca_df <- s_df_sub[,c(14:17,21:23,29)]

## run the adonis
#s_sub_pca_mod <- adonis2(sid_sub_pca_df ~ reef * ftemp * fpco2 * domSymb, data = s_df, method = 'eu', permutations = 1000) # below is the model with non sig interactions removed:
s_sub_pca_mod <- adonis2(sid_sub_pca_df ~ fpco2 + ftemp + reef + domSymb, data = s_df, method = 'eu', permutations = bootnum)
s_sub_pca_mod # view SSID adonis output

# extract pvalues
s_sub_pval <- s_sub_pca_mod["Pr(>F)"]

# perform principal component analysis (PCA)
s_sub_pca <- prcomp(sid_sub_pca_df, center = TRUE, scale= TRUE)

```

```{r SSID subset PCA plot}

# create labels for p values calculated above
s_sub_pco2_pval <- substitute(italic(P[reef])==p, list(p = format(s_sub_pval[1,1], digits = 2)))
s_sub_temp_pval <- substitute(italic(P[temp])==p, list(p = format(s_sub_pval[2,1], digits = 3)))
s_sub_reef_pval <- substitute(italic(P[pCO[2]])==p, list(p = format(s_sub_pval[3,1], digits = 2)))
s_sub_symb_pval <- substitute(italic(P[symb])==p, list(p = format(s_sub_pval[4,1], digits = 2)))
s_sub_TR_pval <- substitute(italic(P[temp%*%reef])==p, list(p = format(s_sub_pval[5,1], digits = 3)))
s_sub_TS_pval <- substitute(italic(P[symb%*%temp])==p, list(p = format(s_sub_pval[6,1], digits = 2)))
s_sub_PS_pval <- substitute(italic(P[symb%*%pCO[2]])==p, list(p = format(s_sub_pval[7,1], digits = 2)))


## PCA plots
# symbiont
s_sub_symb_pca <- autoplot(s_sub_pca, data = s_df_sub, 
         colour = "domSymb",
         #fill = "domSymb",
         size = 3,
         shape = "domSymb",
         frame = TRUE, 
         frame.type = "t", # displaying ellipses with multivariate t-distributions for small n producing heavier tails
         frame.level = 0.95, # using 95% CI for all ellipses
         frame.alpha = 0.01,
         loadings = TRUE, 
         loadings.colour = "black", 
         loadings.label = TRUE,
         loadings.label.colour = "black",         
         loadings.label.size = 4, 
         loadings.label.hjust = 2, 
         loadings.label.vjust = 0.85,
         loadings.label.repel = TRUE) +
    scale_shape_manual("",values = c(21, 24), labels = c("Cladocopium spp.", "Durusdinium spp.")) +
    scale_colour_manual("", labels = c("Cladocopium spp.", "Durusdinium spp."), values = c(wes_palette("Royal1", 3:5, type = "continuous"))) +
    #scale_fill_manual("", labels = c("Cladocopium spp.", "Durusdinium spp."), values = c(wes_palette("Royal1", 3:5, type = "continuous"))) +
    guides(fill = FALSE, color = guide_legend(label.theme = element_text(face = "italic", size = 9), keyheight = 0.3, nrow = 2, byrow = TRUE, override.aes = list(linetype = c(0, 0)))) +
    theme_pubr() +
    theme(legend.title = element_blank(), legend.position = c(0.16, 0.95)) +
    annotate("text", x = -0.47, y = -0.38, label = deparse(s_sub_symb_pval), parse = TRUE, size = 4)

# temperature
s_sub_temp_pca <- autoplot(s_sub_pca, data = s_df_sub, 
          colour = "ftemp",
          #shape = "fpco2",
          shape = "ftemp",
          fill = "ftemp",               
          frame = TRUE, 
          frame.type = "t", # displaying ellipses with multivariate t-distributions for small n producing heavier tails
          frame.level = 0.95, # using 95% CI for all ellipses
          frame.alpha = 0.01,
          loadings = TRUE, 
          loadings.colour = "black", 
          loadings.label = TRUE,
          loadings.label.colour = "black",         
          loadings.label.size = 4, 
          loadings.label.hjust = 2, 
          loadings.label.vjust = 0.85,
          loadings.label.repel = TRUE) +
  #scale_shape_manual("pCO2", labels = c("pre industrial", "current", "end-of-century", "extreme"), values = c(21, 23, 24, 22)) +
  scale_shape_manual("temperature", labels = c("28C", "31C"), values = c(21, 21)) +
  scale_colour_manual("temperature", labels = c("28C", "31C"), values = c("#4393c3", "#b2182b"))+
  scale_fill_manual("temperature", labels = c("28C", "31C"), values = c("#4393c3", NA))+
  guides(color = guide_legend(keyheight = 0.3, nrow = 4, byrow = TRUE, override.aes = list(linetype = c(0, 0)))) +
  theme(legend.title = element_blank(), legend.background = element_rect(fill = "transparent", colour = NA)) +
  annotate("text", x = -0.45, y = -0.38, label = deparse(s_sub_temp_pval), parse = TRUE, size = 3)

# pco2
s_sub_pco2_pca <- autoplot(s_sub_pca, data = s_df_sub, 
          colour = "fpco2",
          #shape = "ftemp",
          fill = "fpco2",
          shape = "fpco2",
          frame = TRUE, 
          frame.type = "t", # displaying ellipses with multivariate t-distributions for small n producing heavier tails
          frame.level = 0.95, # using 95% CI for all ellipses
          frame.alpha = 0.01,
          loadings = TRUE, 
          loadings.colour = "black", 
          loadings.label = TRUE,
          loadings.label.colour = "black",         
          loadings.label.size = 4, 
          loadings.label.hjust = 2, 
          loadings.label.vjust = 0.85,
          loadings.label.repel = TRUE) +
  #scale_shape_manual("temperature", labels = c("28 C", "31 C"), values = c(19,21)) +
  scale_shape_manual("", labels = c("pre industrial", "current", "end-of-century", "extreme"), values = c(21, 23, 24, 22)) +
  scale_fill_manual("", labels = c("pre industrial", "current", "end-of-century", "extreme"), values = c("#b2abd2", "#5e3c99", "#fdb863", "#e66101"))+
  scale_colour_manual("", labels = c("pre industrial", "current", "end-of-century", "extreme"), values = c("#b2abd2", "#5e3c99", "#fdb863", "#e66101"))+
  guides(fill = guide_legend(nrow = 2), shape = guide_legend(nrow = 2), color = guide_legend(nrow = 2, override.aes = list(linetype = c(0, 0, 0, 0)))) +
  annotate("text", x = -0.45, y = -0.38, label = deparse(s_sub_pco2_pval), parse = TRUE, size = 3)

# reef
s_sub_reef_pca <- autoplot(s_sub_pca, data = s_df_sub, 
          colour = "reef",
          fill = "reef",
          frame = TRUE, 
          frame.type = "t", # displaying ellipses with multivariate t-distributions for small n producing heavier tails
          frame.level = 0.95, # using 95% CI for all ellipses
          frame.alpha = 0.01,
          loadings = TRUE, 
          loadings.colour = "black", 
          loadings.label = TRUE,
          loadings.label.colour = "black",         
          loadings.label.size = 4, 
          loadings.label.hjust = 2, 
          loadings.label.vjust = 0.85,
          loadings.label.repel = TRUE) +
  scale_colour_manual("reef environment", labels = c("offshore", "inshore"), values = c("#00a08b", "#e9aa2b")) +
  scale_fill_manual("reef environment", labels = c("offshore", "inshore"), values = c("#00a08b", "#e9aa2b")) +
  guides(fill = FALSE, color = guide_legend(keyheight = 0.3, nrow = 2, override.aes = list(linetype = c(0, 0))), shape = guide_legend(nrow = 2, override.aes = list(linetype = c(0, 0)))) +
  theme(legend.title = element_blank(), legend.background = element_rect(fill = "transparent", colour = NA)) +
  annotate("text", x = -0.45, y = -0.38, label = deparse(s_sub_reef_pval), parse = TRUE, size = 3)

```

<br/>
<br/>


### Correlation assessments

**Not sure I want to include this? But maybe put it in the supplemental materials!**

##### *Siderastrea siderea* subset correlations 
```{r SSID subset corr plot}

## create a C dom and D dom dataframe
sid_corr_D <- s_df_sub[s_df_sub$domSymb == "D",] 
sid_corr_C <- s_df_sub[s_df_sub$domSymb == "C",]

## Create new correlation dataframe for SSID subset
sid_sub_corr_df <- s_df_sub[,c(10:11,14:17,21:23,30)] %>% 
  mutate(colour = case_when(domSymb == "C" ~ "#899DA4",
                           domSymb == "D" ~ "#E19171"),
         shape = case_when(fpco2 == "300" ~ "21",
                            fpco2 == "420" ~ "23",
                            fpco2 == "680" ~ "24",
                            fpco2 == "3300" ~ "22"))


## Correlation matrix and scatterplots coloured by dominant symbiont type for all subset samples
corrgram(sid_sub_corr_df, order = FALSE, lower.panel = panel.fill.R2, upper.panel = panel.pts.col, text.panel = panel.txt.brdr,
         col.regions = colorRampPalette(c("#ffffcc", "#c7e9b4", "#225ea8")), 
         color = sid_sub_corr_df$colour, pch = as.numeric(as.character(sid_sub_corr_df$shape)),
         labels = c("symbiont \ndensity", "host \nprotein", "chlorophyll a", "colour \nintensity", "calcification \nrate", "host \ncarbohydrate", "host \nlipid")) 

```

##### *Siderastrea siderea* dominated by *Cladocopium spp.*  correlations 
```{r SSID C subset corr plot}

## Create new correlation dataframe for SSID C subset
sid_sub_corr_C_df <- sid_corr_C[,c(10:11,14:17,21:23)] %>% 
  mutate(shape = case_when(ftemp == "28" ~ "19",
                           ftemp == "31" ~ "21"),
         colour = case_when(fpco2 == "300" ~ "#b2abd2",
                            fpco2 == "420" ~ "#5e3c99",
                            fpco2 == "680" ~ "#fdb863",
                            fpco2 == "3300" ~ "#e66101"))


## Correlation matrix and scatterplots coloured by pCO2 for C dominated samples only
corrgram(sid_sub_corr_C_df, order = FALSE, lower.panel = panel.fill.R2, upper.panel = panel.pts.col, text.panel = panel.txt.brdr,
         col.regions = colorRampPalette(c("#ffffcc", "#c7e9b4", "#225ea8")), 
         color = sid_sub_corr_C_df$colour, pch = as.numeric(as.character(sid_sub_corr_C_df$shape)),
         labels = c("symbiont \ndensity", "host \nprotein", "chlorophyll a", "colour \nintensity", "calcification \nrate", "host \ncarbohydrate", "host \nlipid")) 

```

##### *Siderastrea siderea* dominated by *Durusdinium spp.*  correlations 
```{r SSID D subset corr plot}

## Create new correlation dataframe for SSID D subset
sid_sub_corr_D_df <- sid_corr_D[,c(10:11,14:17,21:23)] %>% 
  mutate(shape = case_when(ftemp == "28" ~ "19",
                           ftemp == "31" ~ "21"),
         colour = case_when(fpco2 == "300" ~ "#b2abd2",
                            fpco2 == "420" ~ "#5e3c99",
                            fpco2 == "680" ~ "#fdb863",
                            fpco2 == "3300" ~ "#e66101"))


## Correlation matrix and scatterplots coloured by pCO2 for D dominated samples only
corrgram(sid_sub_corr_D_df, order = FALSE, lower.panel = panel.fill.R2, upper.panel = panel.pts.col, text.panel = panel.txt.brdr,
         col.regions = colorRampPalette(c("#ffffcc", "#c7e9b4", "#225ea8")), 
         color = sid_sub_corr_D_df$colour, pch = as.numeric(as.character(sid_sub_corr_D_df$shape)),
         labels = c("symbiont \ndensity", "host \nprotein", "chlorophyll a", "colour \nintensity", "calcification \nrate", "host \ncarbohydrate", "host \nlipid")) 

```

<br/>
<br/>


### Plasticity analysis

```{r SSID subset distance dataframe setup}

s_sub_dis <- s_sub_pca$x[,1:2] # grab PC1 and PC2 distances from prcomp() object
sid_sub_dist <- cbind(s_df_sub[,c(1,7,10,11,12,27,30)], s_sub_dis) # combine the datasets

# adds column with the control value PC1/PC2 per colony
sid_sub_dist$cont_pc1[sid_sub_dist$treat2 == "447_28"] <- sid_sub_dist$PC1[sid_sub_dist$treat2 == "447_28"]
sid_sub_dist$cont_pc2[sid_sub_dist$treat2 == "447_28"] <- sid_sub_dist$PC2[sid_sub_dist$treat2 == "447_28"]

# get the 400_28 PC values
sid_sub_dist2 <- sid_sub_dist %>%  
  group_by(colony) %>% 
  summarise(con2_pc1 = sum(cont_pc1, na.rm = TRUE),
            con2_pc2 = sum(cont_pc2, na.rm = TRUE))

# Make columns of the 400_28 PC values
sid_sub_dist <- sid_sub_dist %>% 
  left_join(sid_sub_dist2, by = c("colony" = "colony"))

sid_sub_dist$dist <- sqrt(((sid_sub_dist$PC1 - sid_sub_dist$con2_pc1)^2) + ((sid_sub_dist$PC2 - sid_sub_dist$con2_pc2)^2)) # formula for calculating distance between control (T90) and treatment values
sid_sub_dist$treat_plot <- paste(sid_sub_dist$fpco2, sid_sub_dist$ftemp, sep = "_")


## modify dataframe to remove the control treatment and reorder levels
sid_sub_dist <- sid_sub_dist %>% 
  filter(treat_plot != "420_28") %>% 
  mutate(treat_plot = factor(treat_plot, levels = c("300_28", "300_31", "420_28", "420_31", "680_28", "680_31", "3300_28", "3300_31")),
         reef = factor(reef, levels = c("N", "F")),
         ftemp = factor(ftemp, levels = c("28", "31")),
         fpco2 = factor(fpco2, levels = c("300", "420", "680", "3300")))

```

```{r SSID subset distance analysis, include=FALSE}

## Model selection (via AIC)
ssid_sub_dist_mod <- glm(dist ~ reef * fpco2 * ftemp * domSymb, family = Gamma(link = "log"), data = sid_sub_dist)
ssid_sub_dist_mod2 <- glm(dist ~ reef * fpco2 + ftemp + domSymb, family = Gamma(link = "log"), data = sid_sub_dist)
ssid_sub_dist_mod3 <- glm(dist ~ reef + fpco2 * ftemp + domSymb, family = Gamma(link = "log"), data = sid_sub_dist)
ssid_sub_dist_mod4 <- glm(dist ~ reef + fpco2 + ftemp + domSymb, family = Gamma(link = "log"), data = sid_sub_dist)
ssid_sub_dist_mod5 <- glm(dist ~ domSymb * fpco2 + ftemp + reef, family = Gamma(link = "log"), data = sid_sub_dist)
ssid_sub_dist_mod6 <- glm(dist ~ fpco2 + ftemp + domSymb, family = Gamma(link = "log"), data = sid_sub_dist) # best fit

# check for best-fit model
compare_performance(ssid_sub_dist_mod, ssid_sub_dist_mod2, ssid_sub_dist_mod3, ssid_sub_dist_mod4, ssid_sub_dist_mod5, ssid_sub_dist_mod6, rank = TRUE)
plot(compare_performance(ssid_sub_dist_mod, ssid_sub_dist_mod2, ssid_sub_dist_mod3, ssid_sub_dist_mod4, ssid_sub_dist_mod5, ssid_sub_dist_mod6, rank = TRUE))


## Best-fit GLM with Gamma log link
ssid_sub_dist_glm <- glm(dist ~ fpco2 + ftemp + domSymb, family = Gamma(link = "log"), data = sid_sub_dist, contrasts = list(domSymb=contr.sum, fpco2=contr.sum, ftemp=contr.sum))
summary(ssid_sub_dist_glm) # summary output of the GLM
ssid_sub_anova_out <- Anova(ssid_sub_dist_glm, type = "III", test.statistic = "F") # comparison of the 'main effects'

```

```{r plots of SSID subset distances}

## Summary data for plotting based on mean/SE instead of boot
sid_sub_dist_sum <- sid_sub_dist %>% 
  mutate(fpco2 = factor(fpco2, levels = c("300", "420", "680", "3300"))) %>% 
  #mutate(treat_plot = factor(treat_plot, levels = c("300_28", "300_31", "420_28", "420_31", "680_28", "680_31", "3300_28", "3300_31"))) %>% 
  group_by(fpco2, domSymb) %>% 
  summarise(mean = mean(dist),
            se = (sd(dist) / sqrt(n()))) 


## Plasticity plot
ssid_sub_plast_plot <- ggplot(sid_sub_dist_sum, aes(x = domSymb, y = mean, shape = fpco2, fill = fpco2)) +
  theme_pubr() +
  theme(legend.title = element_blank(), legend.position = c(0.09, 0.87), axis.text.x = element_text(face = "italic"), axis.ticks.x = element_blank()) +
  guides(shape = guide_legend(ncol = 1)) +
  geom_point(data = sid_sub_dist, aes(x = domSymb, y = dist), size = 2, alpha = 0.7, position = position_dodge(width = 0.2)) +
  geom_errorbar(aes(ymin = mean - se, ymax = mean + se), size = 1, width = 0.1, position = position_dodge(width = 0.2)) +
  geom_point(size = 5, position = position_dodge(width = 0.2)) +
  scale_shape_manual(values = c(21, 23, 24, 22)) +
  scale_fill_manual(values = c("#b2abd2", "#5e3c99", "#fdb863", "#e66101")) + 
  labs(y = "PC distance from control", x = "") +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 10)) +
  scale_x_discrete(labels = c("C" = "Cladocopium spp.", "D" = "Durusdinium spp."))
  

## Model output
summary(ssid_sub_dist_glm) # summary output of the GLM
ssid_sub_anova_out # comparison of the 'main effects'

```

<br/>
<br/>


### **Figure 4**

```{r save SSID subset PCA and plasticity, fig.width = 11, fig.height = 5}

ggarrange(s_sub_symb_pca, ssid_sub_plast_plot, ncol = 2, labels = "AUTO") +
  ggsave("Figures/Final_Figures/Figure4_SSIDPhys.pdf", width = 11, height = 5) +
  ggsave("Figures/Final_Figures/Figure4_SSIDPhys.png", width = 11, height = 5)

```

**Figure 4.** *Siderastrea siderea* coral holobiont physiology based on dominant Symbiodinacaea genus. (**A**) Principal component analysis of all physiological parameters depicted by dominant Symbiodinacaea genus (*Cladocopium spp.* grey; *Durusdinium spp.* pink). Arrows represent significant (p < 0.05) correlation vectors for physiological parameters and ellipses represent 95% confidence based on multivariate t-distributions. (**B**) Assessment of phenotypic plasticity based on dominant Symbiodinacaea genus with higher values represent greater plasticity in coral holobiont samples. *p*CO~2~ treatment is depicted by colour and shape (280 $\mu$atm light purple, circle; 400 $\mu$atm dark purple, diamond; 700 $\mu$atm light orange, triangle; 2800 $\mu$atm dark orange, square).

<br/>
<br/>


### *Figure S2*

```{r Figure S2 SSID subset PCA all plots, fig.height = 4.3, fig.width = 12}

ggarrange(s_sub_pco2_pca, s_sub_temp_pca, s_sub_reef_pca, nrow = 1, ncol = 3, labels = "AUTO", align = "h") +
  ggsave("Figures/Supplemental_Figures/FigureS2_SSID_PCAs.pdf", width = 13, height = 5.1) +
  ggsave("Figures/Supplemental_Figures/FigureS2_SSID_PCAs.png", width = 13, height = 5.1)

```

**Figure S2.** Principal component analysis of *Siderastrea siderea* coral holobiont physiology subset with known Symbiodinacaea community composition depicted by (**A**) *p*CO~2~ (280 $\mu$atm light purple; 400 $\mu$atm dark purple; 700 $\mu$atm light orange; 2800 $\mu$atm dark orange), (**B**) temperature treatment (28$^\circ$ C blue; 31$^\circ$ C red), and (**C**) natal reef environment (offshore green; inshore yellow). Arrows represent significant (p < 0.05) correlation vectors for physiological parameters and ellipses represent 95% confidence based on multivariate t-distributions. 

<br/>
<br/>


## Supplemental Tables{.tabset} 

### Table S1 *WORKING*

**Figure S1.** Akaike information criterion (AIC) generalized linear model (GLM) selection comparisons for plasticity assessments to select the best-fit model per species.
```{r table S1 for markdown}

## combine model output
plast_aic <- rbind(ssid_plast_aic, pstr_plast_aic, past_plast_aic) # combine all PERMANOVA model outputs
plast_aic$species <- c(rep("SSID", 6), rep("PSTR", 6), rep("PAST", 6)) # add column for species ID


##### STILL NEED TO MAKE THIS PRETTIER BUT JUST A PLACE HOLDER FOR NOW

### update formatting of text 
#pca_mod_out <- pca_mod_out %>% 
#  mutate(SumOfSqs = round(as.numeric(SumOfSqs), 0), # round values
#         R2 = round(as.numeric(R2), 3),
#         `F` = round(as.numeric(`F`), 2),
#         `Pr(>F)` = round(as.numeric(`Pr(>F)`), 5)) %>% 
#  mutate(treat = gsub("[0-9]", "", treat), # rename the treatment IDs
#         treat = gsub("fpco", "pCO2", treat),
#         treat = gsub("ftemp", "temperature", treat),
#         treat = gsub("reef", "reef environment", treat)) %>% 
#  rename("P-value" = `Pr(>F)`, # rename columns
#         " " = treat)
#  
## combine them with format
table_s1 <- kable(plast_aic[-2], booktabs = TRUE, row.names = FALSE) %>%
  kable_styling(font_size = 12, full_width = FALSE) %>% 
  pack_rows("Siderastrea Siderea", 1, 6, italic = TRUE) %>%
  pack_rows("Pseudodiploria strigosa", 7, 12, italic = TRUE) %>%
  pack_rows("Porites astreoides", 13, 18, italic = TRUE)
table_s1

```

<br/>
<br/>


---

### Table S2

**Table S2.** PERMANOVA model output from each species using the *adonis2* function with 1500 iterations.
```{r table S2 for markdown}

## combine model output
pca_mod_out <- rbind(s_pca_mod, p_pca_mod, a_pca_mod) # combine all PERMANOVA model outputs
pca_mod_out$treat <- rownames(pca_mod_out) # add column for rownames
pca_mod_out$species <- c(rep("SSID", 5), rep("PSTR", 5), rep("PAST", 5)) # add column for species ID
pca_mod_out <- pca_mod_out[c(6, 1:5, 7)] # reorder columns so treatment is first


## update formatting of text 
pca_mod_out <- pca_mod_out %>% 
  mutate(SumOfSqs = round(as.numeric(SumOfSqs), 0), # round values
         R2 = round(as.numeric(R2), 3),
         `F` = round(as.numeric(`F`), 2),
         `Pr(>F)` = round(as.numeric(`Pr(>F)`), 5)) %>% 
  mutate(treat = gsub("[0-9]", "", treat), # rename the treatment IDs
         treat = gsub("fpco", "pCO2", treat),
         treat = gsub("ftemp", "temperature", treat),
         treat = gsub("reef", "reef environment", treat)) %>% 
  rename("P-value" = `Pr(>F)`, # rename columns
         " " = treat,
         "Sum of Squares" = SumOfSqs)
  
## combine them with format
table_s2 <- kable(pca_mod_out[-7], booktabs = TRUE, row.names = FALSE) %>%
  kable_styling(font_size = 12, full_width = FALSE) %>% 
  pack_rows("Siderastrea Siderea", 1, 5, italic = TRUE) %>%
  pack_rows("Pseudodiploria strigosa", 6, 10, italic = TRUE) %>%
  pack_rows("Porites astreoides", 11, 15, italic = TRUE) %>% 
  row_spec(c(4:5,9:10,14:15), italic = TRUE)
table_s2

```

<br/>
<br/>


---

### Table S3

**Table S3.** GLM output from plasticity assessments for each species. The intercept of each model was set as 300 $\mu$atm, 28$^\circ$C, and inshore reef environment.
```{r table S3 for markdown}

## combine model output per species
dist_mod_out <- rbind(tidy(ssid_dist_glm), tidy(pstr_dist_glm), tidy(past_dist_glm)) # combine all glm model outputs (the use of tidy() makes a tibble from the GLM)
dist_mod_out$species <- c(rep("SSID", 9), rep("PSTR", 6), rep("PAST", 6)) # add column for species ID


## update formatting of text 
dist_mod_out <- dist_mod_out %>% 
  mutate(estimate = round(as.numeric(estimate), 3), # round values
         std.error = round(as.numeric(std.error), 3),
         statistic = round(as.numeric(statistic), 2),
         p.value = round(as.numeric(p.value), 3)) %>% 
  mutate(term = gsub("21", "-current", term), # rename the treatment IDs
         term = gsub("22", "-EOC", term),
         term = gsub("21", "-extreme", term),
         term = gsub("[0-9]", "", term),
         term = gsub("fpco", "pCO2", term),
         term = gsub("ftemp", "temperature (31C)", term),
         term = gsub("reef", "reef environment (offshore)", term)) %>% 
  rename(" " = term, # rename columns
        "Estimate" = estimate,
        "Standard error" = std.error,
        "Statistic" = statistic,
        "P-value" = p.value)
  
## combine them with format
table_s3 <- kable(dist_mod_out[-6], booktabs = TRUE, row.names = FALSE) %>%
  kable_styling(font_size = 12, full_width = FALSE) %>% 
  pack_rows("Siderastrea Siderea", 1, 9, italic = TRUE) %>%
  pack_rows("Pseudodiploria strigosa", 10, 15, italic = TRUE) %>%
  pack_rows("Porites astreoides", 16, 21, italic = TRUE)
table_s3

```

<br/>
<br/>


---

### Table S4

**Table S4.** Type III test of analysis of deviance output based on the plasticity GLM per species in **Table S3**.
```{r table S4 for markdown}

## combine ANOVA output per species
anova_mod_out <- rbind(ssid_anova_out, pstr_anova_out, past_anova_out) # combine all ANOVA model outputs
anova_mod_out$treat <- rownames(anova_mod_out) # add column for rownames
anova_mod_out$species <- c(rep("SSID", 5), rep("PSTR", 4), rep("PAST", 4)) # add column for species ID
anova_mod_out <- anova_mod_out[c(5, 1:4, 6)] # reorder columns so treatment is first


## update formatting of text 
anova_mod_out <- anova_mod_out %>% 
  mutate(`Sum Sq` = round(as.numeric(`Sum Sq`), 2), # round values
         `F values` = round(as.numeric(`F values`), 1),
         `Pr(>F)` = round(as.numeric(`Pr(>F)`), 2)) %>% 
  mutate(treat = gsub("[0-9]", "", treat), # rename the treatment IDs
         treat = gsub("fpco", "pCO2", treat),
         treat = gsub("ftemp", "temperature", treat),
         treat = gsub("reef", "reef environment", treat)) %>% 
  rename("P-value" = `Pr(>F)`, # rename columns
         " " = treat,
         "Sum of Squares" = `Sum Sq`)
  
## combine them with format
table_s4 <- kable(anova_mod_out[-6], booktabs = TRUE, row.names = FALSE) %>%
  kable_styling(font_size = 12, full_width = FALSE) %>% 
  pack_rows("Siderastrea Siderea", 1, 5, italic = TRUE) %>%
  pack_rows("Pseudodiploria strigosa", 6, 9, italic = TRUE) %>%
  pack_rows("Porites astreoides", 10, 13, italic = TRUE) %>% 
  row_spec(c(5,9,13), italic = TRUE)
table_s4

```

<br/>
<br/>


---

### Table S5

**Table S5.** PERMANOVA model output from the *Siderastrea siderea* samples with known Symbiodiniaceae community composition using the *adonis2* function with 1500 iterations.
```{r table S5 for markdown}

## rearrange SSID PCA output dataframe
sub_pca_out <- s_sub_pca_mod # just renaming the SSID PCA output
sub_pca_out$treat <- rownames(s_sub_pca_mod) # add column for rownames
sub_pca_out <- sub_pca_out[c(6, 1:5)] # reorder columns so treatment is first


## update formatting of text 
sub_pca_out <- sub_pca_out %>% 
  mutate(SumOfSqs = round(as.numeric(SumOfSqs), 0), # round values
         R2 = round(as.numeric(R2), 3),
         `F` = round(as.numeric(`F`), 2),
         `Pr(>F)` = round(as.numeric(`Pr(>F)`), 5)) %>% 
  mutate(treat = gsub("[0-9]", "", treat), # rename the treatment IDs
         treat = gsub("fpco", "pCO2", treat),
         treat = gsub("ftemp", "temperature", treat),
         treat = gsub("reef", "reef environment", treat),
         treat = gsub("domSymb", "dominant symbiont", treat)) %>% 
  rename("P-value" = `Pr(>F)`, # rename columns
         " " = treat,
         "Sum of Squares" = SumOfSqs)
  
## combine them with format
table_s5 <- kable(sub_pca_out, booktabs = TRUE, row.names = FALSE) %>%
  kable_styling(font_size = 12, full_width = FALSE) %>% 
  row_spec(5:6, italic = TRUE)
table_s5

```

<br/>
<br/>


---

### Table S6

**Table S6.** GLM output from plasticity assessments for the *Siderastrea siderea* samples with known Symbiodiniaceae community composition. The intercept of each model was set as 300 $\mu$atm, 28$^\circ$C, and *Cladocopium spp.* community dominance.
```{r table S6 for markdown}

## update formatting of text 
ssid_dist_mod_out <- tidy(ssid_sub_dist_glm) %>% 
  mutate(estimate = round(as.numeric(estimate), 3), # round values
         std.error = round(as.numeric(std.error), 3),
         statistic = round(as.numeric(statistic), 2),
         p.value = round(as.numeric(p.value), 3)) %>% 
  mutate(term = gsub("21", "-current", term), # rename the treatment IDs
         term = gsub("22", "-EOC", term),
         term = gsub("21", "-extreme", term),
         term = gsub("[0-9]", "", term),
         term = gsub("fpco", "pCO2", term),
         term = gsub("ftemp", "temperature (31C)", term),
         term = gsub("domSymb", "dominant symbiont (D)", term)) %>% 
  rename(" " = term, # rename columns
        "Estimate" = estimate,
        "Standard error" = std.error,
        "Statistic" = statistic,
        "P-value" = p.value)
  
## combine them with format
table_s6 <- kable(ssid_dist_mod_out, booktabs = TRUE, row.names = FALSE) %>%
  kable_styling(font_size = 12, full_width = FALSE)
table_s6

```

<br/>
<br/>


---

### Table S7

**Table S7.** Type III test of analysis of deviance output based on the plasticity GLM for the *Siderastrea siderea* samples with known Symbiodiniaceae community composition in **Table S6**.
```{r table S7 for markdown}

## modify ANOVA output
anova_sub_mod_out <- ssid_sub_anova_out # rename the ANOVA output
anova_sub_mod_out$treat <- rownames(anova_sub_mod_out) # add column for rownames
anova_sub_mod_out <- anova_sub_mod_out[c(5, 1:4)] # reorder columns so treatment is first


## update formatting of text 
anova_sub_mod_out <- anova_sub_mod_out %>% 
  mutate(`Sum Sq` = round(as.numeric(`Sum Sq`), 2), # round values
         `F values` = round(as.numeric(`F values`), 1),
         `Pr(>F)` = round(as.numeric(`Pr(>F)`), 2)) %>% 
  mutate(treat = gsub("[0-9]", "", treat), # rename the treatment IDs
         treat = gsub("fpco", "pCO2", treat),
         treat = gsub("ftemp", "temperature", treat),
         treat = gsub("domSymb", "dominant symbiont", treat)) %>% 
  rename("P-value" = `Pr(>F)`, # rename columns
         " " = treat,
         "Sum of Squares" = `Sum Sq`)
  
## combine them with format
table_s7 <- kable(anova_sub_mod_out, booktabs = TRUE, row.names = FALSE) %>%
  kable_styling(font_size = 12, full_width = FALSE) %>% 
  row_spec(4, italic = TRUE)
table_s7

```

<br/>
<br/>


---

<br/>

```{r excel WB, message=FALSE, warning=FALSE}

### Create workbook
wb <- createWorkbook()

# add 'Table S1' worksheet
addWorksheet(wb, "Table S1 - Plasticity AIC") 
writeData(wb, sheet = 1, x = plast_aic)
setColWidths(wb, sheet = 1, cols = 1:7, widths = "auto")

# add 'Table S2' worksheet
addWorksheet(wb, "Table S2 - PERMANOVA all") 
writeData(wb, sheet = 2, x = pca_mod_out)
setColWidths(wb, sheet = 2, cols = 1:7, widths = "auto")

# add 'Table S3' worksheet
addWorksheet(wb, "Table S3 - Plasticity GLM") 
writeData(wb, sheet = 3, x = dist_mod_out)
setColWidths(wb, sheet = 3, cols = 1:6, widths = "auto")

# add 'Table S4' worksheet
addWorksheet(wb, "Table S4 - Plasticity ANOVA") 
writeData(wb, sheet = 4, x = anova_mod_out)
setColWidths(wb, sheet = 4, cols = 1:6, widths = "auto")

# add 'Table S5' worksheet
addWorksheet(wb, "Table S5 - SSID PERMANOVA") 
writeData(wb, sheet = 5, x = sub_pca_out)
setColWidths(wb, sheet = 5, cols = 1:6, widths = "auto")

# add 'Table S6' worksheet
addWorksheet(wb, "Table S6 - SSID plasticity GLM") 
writeData(wb, sheet = 6, x = ssid_dist_mod_out)
setColWidths(wb, sheet = 6, cols = 1:5, widths = "auto")

# add 'Table S7' worksheet
addWorksheet(wb, "Table S7 - SSID plast ANOVA") 
writeData(wb, sheet = 7, x = anova_sub_mod_out)
setColWidths(wb, sheet = 7, cols = 1:5, widths = "auto")

# save workbook
saveWorkbook(wb, file="Data/Supplemental/Supplemental_Tables.xlsx", overwrite = TRUE)


```


## Session information

Session information from the last run date on `r date`:

```{r}

sessionInfo()

```


